src/main/java/lery/Storage.java

```diff
@@ -0,0 +1,150 @@
+                    }
+                    if (type.equalsIgnoreCase("T")) {
+                        newTask = new Todo(event);
+                    } else if (type.equalsIgnoreCase("D")) {
+                        String extraInfo = splitLine[3];
+                        checkDateFormat(extraInfo);
+
+                        newTask = new Deadline(event, extraInfo);
+                    } else if (type.equalsIgnoreCase("E")) {
+                        String extraInfo = splitLine[3];
+                        newTask = new Event(event, extraInfo);
+                    } else {
+                        throw new LeryException("Erm... Invalid type!" + type);
+                    }
+                    this.taskList.add(newTask);
+
+                }
+                s.close();
+            }
+            return this.taskList;
+        } catch (IOException | ArrayIndexOutOfBoundsException e) {
+            throw new LeryException("Erm... Error loading tasks from file");
+        }
+    }
+
+    /**
+     * Saves a new task to the text file and adds it to the task list.
+     *
+     * @param newTask the task to be saved.
+     * @throws LeryException if an error occurs during the saving process.
+     */
+    public void saveTasks(Task newTask) throws LeryException {
+        String msg = " | 0 | ";
+        msg = "\n" + newTask.getType() + msg + newTask.getDescription();
+        if (newTask instanceof Event || newTask instanceof Deadline) {
+            msg += ("| " + newTask.getExtraInfoShortened());
+        }
+        try {
+            FileWriter fw = new FileWriter(filename, true);
+            fw.write(msg);
+            fw.close();
+            this.taskList.add(newTask);
+
+        } catch (IOException e) {
+            throw new LeryException("Erm... Error saving tasks to file");
+        }
+    }
+
+    /**
+     * Checks if the given date in the command string is in the correct format (yyyy-MM-dd).
+     *
+     * @param date the date string to be checked.
+     * @throws LeryException if the date is not in the correct format.
+     */
+    public void checkDateFormat(String date) throws LeryException {
+        try {
+            LocalDate d = LocalDate.parse(date, this.formatter);
+        } catch (Exception ex) {
+
+            throw new LeryException("Erm... Date not keyed in correct format! Correct format is yyyy-MM-dd" + ex);
+
+        }
+    }
+
+    /**
+     * Retrieves the current task list.
+     *
+     * @return the TaskList containing tasks.
+     */
+    public TaskList getTaskList() {
+        return this.taskList;
+    }
+
+    /**
+     * Gets the number of tasks in the task list.
+     *
+     * @return the size of the task list.
+     */
+    public int getSize() {
+        return this.taskList.getSize();
+    }
+
+    /**
+     * Deletes a specified task from the task list.
+     *
+     * @param t the task to be deleted.
+     */
+    public void delete(Task t) {
```

Comment:
Perhaps deleteTask would be more apt, following your previous naming conventions 😄 
```suggestion
    public void deleteTask(Task t) {
```